### PR TITLE
DOC: Update the pandas.core.window.x.sum docstring

### PR DESCRIPTION
**Pull request by othersideoflearning**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x ] PR title is "DOC: update the <your-function-or-method> docstring"
- [x ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x ] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
# paste output of "scripts/validate_docstrings.py <your-function-or-method>" here
# between the "```" 
(pandas-dev) D:\GitHub\pandas>python scripts/validate_docstrings.py pandas.core.window.Rolling.sum

################################################################################
################## Docstring (pandas.core.window.Rolling.sum) ##################
################################################################################

Calculate rolling sum of given DataFrame or Series.

Parameters
----------
*args
    Under Review.
**kwargs
    Under Review.

Returns
-------
Series or DataFrame
    Like-indexed object containing the result of function application

See Also
--------
Series.rolling : Calling object with Series data
DataFrame.rolling  : Calling object with DataFrame data
Series.sum  : Equivalent method for Series
DataFrame.sum   : Equivalent method for DataFrame

Examples
--------
>>> s = pd.Series([1, 2, 3, 4, 5])
>>> s.rolling(3).sum()
0     NaN
1     NaN
2     6.0
3     9.0
4    12.0
dtype: float64
>>> s.expanding(3).sum()
0     NaN
1     NaN
2     6.0
3    10.0
4    15.0
dtype: float64
>>> s.rolling(3, center=True).sum()
0     NaN
1     6.0
2     9.0
3    12.0
4     NaN
dtype: float64

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
        No extended summary found
        Errors in parameters section
                Parameters {'kwargs', 'args'} not documented
                Unknown parameters {'**kwargs', '*args'}
                Parameter "*args" has no type
                Parameter "**kwargs" has no type

(pandas-dev) D:\GitHub\pandas>python scripts/validate_docstrings.py pandas.core.window.Expanding.sum

################################################################################
################# Docstring (pandas.core.window.Expanding.sum) #################
################################################################################

Calculate expanding sum of given DataFrame or Series.

Parameters
----------
*args
    Under Review.
**kwargs
    Under Review.

Returns
-------
Series or DataFrame
    Like-indexed object containing the result of function application

See Also
--------
Series.expanding : Calling object with Series data
DataFrame.expanding  : Calling object with DataFrame data
Series.sum  : Equivalent method for Series
DataFrame.sum   : Equivalent method for DataFrame

Examples
--------
>>> s = pd.Series([1, 2, 3, 4, 5])
>>> s.rolling(3).sum()
0     NaN
1     NaN
2     6.0
3     9.0
4    12.0
dtype: float64
>>> s.expanding(3).sum()
0     NaN
1     NaN
2     6.0
3    10.0
4    15.0
dtype: float64
>>> s.rolling(3, center=True).sum()
0     NaN
1     6.0
2     9.0
3    12.0
4     NaN
dtype: float64

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
        No extended summary found
        Errors in parameters section
                Parameters {'kwargs', 'args'} not documented
                Unknown parameters {'**kwargs', '*args'}
                Parameter "*args" has no type
                Parameter "**kwargs" has no type

(pandas-dev) D:\GitHub\pandas>python scripts/validate_docstrings.py pandas.core.window.Window.sum

################################################################################
################## Docstring (pandas.core.window.Window.sum)  ##################
################################################################################

Calculate window sum of given DataFrame or Series.

Parameters
----------
*args
    Under Review.
**kwargs
    Under Review.

Returns
-------
Series or DataFrame
    Like-indexed object containing the result of function application

See Also
--------
Series.window : Calling object with Series data
DataFrame.window  : Calling object with DataFrame data
Series.sum  : Equivalent method for Series
DataFrame.sum   : Equivalent method for DataFrame

Examples
--------
>>> s = pd.Series([1, 2, 3, 4, 5])
>>> s.rolling(3).sum()
0     NaN
1     NaN
2     6.0
3     9.0
4    12.0
dtype: float64
>>> s.expanding(3).sum()
0     NaN
1     NaN
2     6.0
3    10.0
4    15.0
dtype: float64
>>> s.rolling(3, center=True).sum()
0     NaN
1     6.0
2     9.0
3    12.0
4     NaN
dtype: float64

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
        No extended summary found
        Errors in parameters section
                Parameters {'args', 'kwargs'} not documented
                Unknown parameters {'*args', '**kwargs'}
                Parameter "*args" has no type
                Parameter "**kwargs" has no type


```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/core/window.py [hunk 1]
```diff
@@ -320,7 +320,52 @@ def aggregate(self, arg, *args, **kwargs):
...
+    ----------
+    *args
+        Under Review.
+    **kwargs
+        Under Review.
+
+    Returns
+    -------
+    Series or DataFrame
+        Like-indexed object containing the result of function application
```
Comment (WillAyd) [inline pandas/core/window.py]: Period at end

File: pandas/core/window.py [hunk 2]
```diff
@@ -320,7 +320,52 @@ def aggregate(self, arg, *args, **kwargs):
...
+    4    12.0
+    dtype: float64
+    >>> s.expanding(3).sum()
+    0     NaN
+    1     NaN
+    2     6.0
+    3    10.0
+    4    15.0
+    dtype: float64
+    >>> s.rolling(3, center=True).sum()
```
Comment (WillAyd) [inline pandas/core/window.py]: I think this is a great example, but maybe add one line or comment for what to expect with `center=True`

File: pandas/core/window.py [hunk 3]
```diff
@@ -320,7 +320,52 @@ def aggregate(self, arg, *args, **kwargs):
...
+    Returns
+    -------
+    Series or DataFrame
+        Like-indexed object containing the result of function application
+
+    See Also
+    --------
+    Series.%(name)s : Calling object with Series data
+    DataFrame.%(name)s  : Calling object with DataFrame data
+    Series.sum  : Equivalent method for Series
```
Comment (WillAyd) [inline pandas/core/window.py]: Whitespace seems off on this line and below - are there more than one before the ":"?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/window.py [hunk 4]
```diff
@@ -320,7 +320,52 @@ def aggregate(self, arg, *args, **kwargs):
...
+    --------
+    >>> s = pd.Series([1, 2, 3, 4, 5])
+    >>> s.rolling(3).sum()
+    0     NaN
+    1     NaN
+    2     6.0
+    3     9.0
+    4    12.0
+    dtype: float64
+    >>> s.expanding(3).sum()
```
Comment (jreback) [inline pandas/core/window.py]: blank line between cases

File: pandas/core/window.py [hunk 5]
```diff
@@ -639,7 +684,7 @@ def aggregate(self, arg, *args, **kwargs):
     agg = aggregate
 
     @Substitution(name='window')
-    @Appender(_doc_template)
+#    @Appender(_doc_template)
```
Comment (jreback) [inline pandas/core/window.py]: don't comment, simply remove

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window.py (hunk 5 above)
Comment (apallekonda) [reply pandas/core/window.py]: Your comments are valuable and I have deleted necessary code.

File: pandas/core/window.py (hunk 2 above)
Comment (apallekonda) [reply pandas/core/window.py]: Thank you. I have added one line of explanation stating that how this argument helps.

File: pandas/core/window.py (hunk 4 above)
Comment (apallekonda) [reply pandas/core/window.py]: I have added one line for each case.

File: pandas/core/window.py (hunk 3 above)
Comment (apallekonda) [reply pandas/core/window.py]: yup. Deleted.

File: pandas/core/window.py (hunk 1 above)
Comment (apallekonda) [reply pandas/core/window.py]: Done.

**Review by othersideoflearning — COMMENTED** [review]
Added or deleted necessary code as per comments.

**Comment by TomAugspurger** [comment]
Thanks @apallekonda !

**Pull request merged**